bpfman-operator/apis/v1alpha1/cluster_fentry_program_types.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2024 The bpfman Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All fields are required unless explicitly marked optional
+// +kubebuilder:validation:Required
+package v1alpha1
+
+// ClFentryProgramInfo defines the Fentry program details
+type ClFentryProgramInfo struct {
+	ClFentryLoadInfo `json:",inline"`
+	// Whether the program should be attached to the function.
+	// +optional
+	// +kubebuilder:validation:MaxItems=1
+	// +kubebuilder:default:={}
+	Links []ClFentryAttachInfo `json:"links"`
+}
+
+// ClFentryLoadInfo contains the program-specific load information for Fentry
+// programs
+type ClFentryLoadInfo struct {
+	// function is the name of the function to attach the Fentry program to.
+	Function string `json:"function"`
```

Comment:
Can this be any arbitrary string value? Any validation you can put in place?

Reply:
not really it is an ebpf program name so it can be any string

Reply:
so something like `!@ASjdhflaksjdfy@&!^*#@&^!` would be valid?

Reply:
Are there any conventions, standards, or validations that exist _somewhere_ that enforces a particular pattern for the function names?

Reply:
If not a particular pattern, is there any reasonable upper bound for the length of the function name?

Reply:
Rules for C identifers apply here. So must be `^[a-zA-Z][a-zA-Z0-9_]+`. 64 is a reasonable upperbound.